packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -843,7 +868,23 @@ export function hydrateHoistable(
   internalInstanceHandle: Object,
 ): Instance {
   const ownerDocument = getDocumentFromRoot(hoistableRoot);
-  const nodes = ownerDocument.getElementsByTagName(type);
+  const hoistedTagsCache = getHoistedTagsCache(ownerDocument);
+
+  let cache = hoistedTagsCache[type];
+  if (cache === null) {
+    // The cache gets reset on every commit. We only want to query the set of valid
+    //
+    cache = hoistedTagsCache[type] = new Set();
+
+    const nodes = ownerDocument.getElementsByTagName(type);
+    for (let i = 0; i < nodes.length; i++) {
+      const node = nodes[i];
+      if (!isOwnedInstance(node) && node.namespaceURI !== SVG_NAMESPACE) {
+        cache.add(node);
+      }
+    }
```

Comment:
I may want to fork this for `link` so i can exclude `data-precedence` nodes too. They will never be 'hoistable' and there may be many of them in the document. Better yet I may want to use `querySelectorAll('link:not([data-precedence])')` but I need to see if JSDOM will support that correctly in tests